packages/widgetbook_cli/bin/commands/publish.dart

```diff
@@ -0,0 +1,353 @@
+      )
+      ..addOption(
+        'actor',
+        help: 'The username of the actor which triggered the build.',
+      )
+      ..addOption(
+        'git-provider',
+        help: 'The name of the Git provider.',
+        defaultsTo: 'Local',
+        allowed: [
+          'GitHub',
+          'GitLab',
+          'BitBucket',
+          'Azure',
+          'Local',
+        ],
+      )
+      ..addOption(
+        'base-branch',
+        help:
+            'The base branch of the pull-request. For example, main or master.',
+      )
+      ..addOption(
+        'base-commit',
+        help: 'The SHA hash of the commit of the base branch.',
+      )
+      ..addOption(
+        'github-token',
+        help: 'GitHub API token.',
+      )
+      ..addOption(
+        'pr',
+        help: 'The number of the PR.',
+      );
+  }
+
+  @override
+  final String description = 'Publish a new build';
+
+  @override
+  final String name = 'publish';
+
+  late CiParserRunner? ciParserRunner;
+
+  @override
+  Future<int> run() async {
+    final publishProgress = logger.progress(
+      'Uploading build',
+    );
+
+    final path = results['path'] as String;
+
+    if (!await GitDir.isGitDir(path)) {
+      publishProgress.fail();
+      usageException('Directory from "path" is not a Git folder');
+    }
+
+    final gitDir = await GitDir.fromExisting(
+      path,
+      allowSubdirectory: true,
+    );
+
+    publishProgress.update('Obtaining data from Git');
+
+    final isWorkingTreeClean = await gitDir.isWorkingTreeClean();
+
+    final apiKey = results['api-key'] as String;
+    final currentBranch = await gitDir.currentBranch();
+    final branch = results['branch'] as String? ?? currentBranch.branchName;
+
+    final commit = results['commit'] as String? ?? currentBranch.sha;
+
+    final gitProvider = results['git-provider'] as String;
+    final gitHubToken = results['github-token'] as String?;
+    final prNumber = results['pr'] as String?;
+
+    final baseBranch = results['base-branch'] as String?;
+
+    final ciArgsData = CliArgs(
+      apiKey: apiKey,
+      branch: branch,
+      commit: commit,
+      gitProvider: gitProvider,
+      gitHubToken: gitHubToken,
+      prNumber: prNumber,
+      baseBranch: baseBranch,
+      path: path,
+    );
+
+    ciParserRunner = CiParserRunner(
+      argResults: results,
+      gitDir: gitDir,
+    );
+
+    final ciArgs = await ciParserRunner?.getParser()?.getCiArgs();
+
+    if (ciArgs == null) {
+      publishProgress.fail();
+      usageException(
+        'Your CI/CD pipeline provider is currently not supported.',
+      );
+    }
+
+    publishProgress.update('Checking commit');
+    if (!isWorkingTreeClean) {
+      logger
+        ..warn('You have un-commited changes')
+        ..warn('Proceeding with un-committed changes my result to different '
+            'uploads containing the same commit hash, which may cause '
+            'undesired. duplicate use case reviews,');
```

Comment:
This is not proper English. 

What about

> Uploading a new build to Widgetbook Cloud requires a commit SHA. Due to un-committed changes, we are using the commit SHA of your previous commit which can lead to the build being rejected due to an already existing build. 

And then we can print

> Would you like to proceed anyways?

Just FYI: Right now, the build is not being rejected but I think it should be the behaviour of the backend since it does not make any sense two have to builds with the same SHA (since they should be the same)